### PR TITLE
Swift 4 support + Update the Travis CI config.

### PR DESCRIPTION
**Pull request by andersio**
There are a lot of force unwrapping in Objective-C runtime API call sites. A majority of them is previously IUO.

While ideally we can wipe 3.2 implementations away, #3519 and #3525 are motivations to keep them, at least until CocoaPods does their work. 

#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

**Comment by andersio** [comment]
Ping @ReactiveCocoa/reactivecocoa

**Comment by liscio** [comment]
Thanks for undertaking this, @andersio. Looks like it's not as much of a huge transition as I feared it'd be (with all the API changes on macOS, specifically.)

I'm left with a question, though—why not set SWIFT_VERSION = 4.0 in the `.xcodeproj` file along with these changes? Is it just a band-aid for supporting the language until the `${RAC_MAJOR_VERSION}` that is officially updated for Swift 4 and beyond?

**Comment by andersio** [comment]
@liscio Good point. Setting it to 4.0 should do no harm.

**Comment by andersio** [comment]
Argh. SWIFT_VERSION=4.0 is not working for Xcode 8.3.

**Comment by liscio** [comment]
@andersio Argh. This further reinforces my frustration in Xcode's inability to override build settings at the workspace level… 😠

**Comment by andersio** [comment]
@liscio We can drop Xcode 8 support though. Apparently there is no point to continue Xcode 8 support at all.

File: .travis.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       - gem install cocoapods
       - pod repo update
       - pod lib lint ReactiveCocoa.podspec
-      - pod lib lint ReactiveMapKit.podspec
+      - pod lib lint ReactiveMapKit.podspec --swift-version=3.2
```
Comment (andersio) [inline .travis.yml]: This can be removed once the Swift 4 code gets into a release.

**Pull request merged**